Clone.xs

```diff
@@ -327,12 +327,12 @@ sv_clone (SV * ref, HV* hseen, int depth)
     {
       TRACEME(("clone = 0x%x(%d)\n", clone, SvREFCNT(clone)));
       SvREFCNT_dec(SvRV(clone));
-      SvRV(clone) = sv_clone (SvRV(ref), hseen, depth); /* Clone the referent */
-      if (sv_isobject (ref))
-      {
+      SvRV(clone) = sv_clone(SvRV(ref), hseen, depth); /* Clone the referent */
+      if (sv_isobject (ref)) {
           sv_bless (clone, SvSTASH (SvRV (ref)));
       }
       if (SvWEAKREF(ref)) {
+          if (SvREFCNT(SvRV(clone)) <= 2) SvREFCNT_inc(SvRV(clone)); /* weaken is going to decrease it... */
```

Comment:
This is the actual change, right? Are you positive the check is `<= 2`? What is "2"? Can we change it to something more descriptive? What are the chances of "2" being replaced by 1 or 3 in upcoming perl releases? Is `sv_rvweaken()` a public API guaranteed to always decrease the SvREFCNT? Why are we doing this *only* when the ref count is <= 2? Shouldn't we do it every time `SvWEAKREF(ref)` is true?

Reply:
I definitively agree with your comments, and to my mind that '2' thing make everything work but seem a poor fix to me. I'm pretty sure I can find example where it would not be enough, and if this is the case then it would prove that this is a bad solution.

That's why I've extracted the test improvement as its own Pull Request, as I really need to spend more time on this one.

Here are the issues:
- always bumping the refcnt here would make some objects immortal and postpone it to global destruction
- but without bumping it, the weaken object is going to be destroy earlier...

I should have marked it as "Do not Merge Yet", this need more work